src/brogue/Combat.c

```diff
@@ -510,30 +510,34 @@ boolean forceWeaponHit(creature *defender, item *theItem) {
             otherMonster = NULL;
             strcpy(buf2, tileCatalog[pmap[defender->xLoc + newLoc[0] - oldLoc[0]][defender->yLoc + newLoc[1] - oldLoc[1]].layers[highestPriorityLayer(defender->xLoc + newLoc[0] - oldLoc[0], defender->yLoc + newLoc[1] - oldLoc[1], true)]].description);
         }
+        
+        // Only deal force damage if target collides with solid object
+        if (otherMonster || !(tileCatalog[pmap[defender->xLoc + newLoc[0] - oldLoc[0]][defender->yLoc + newLoc[1] - oldLoc[1]].layers[DUNGEON]].flags & (T_LAVA_INSTA_DEATH | T_AUTO_DESCENT | T_IS_DEEP_WATER))) {
```

Comment:
From the comments in the tile catalog in Globals.c, it seems lava and deep water are on the LIQUID layers. Have you tested this? What about using `terrainFlags` macro? 

Additionally, it seems like the `defender->xLoc + newLoc[0] - oldLoc[0]` refers to the location of the next cell in the monsters travel path, if it could continue - so would we not still get the false message if the monster stops on the edge of chasms and lava pools?

Another thing I'm not sure about.. doesn't the check `distanceBetween(oldLoc[0], oldLoc[1], defender->xLoc, defender->yLoc) < weaponForceDistance(netEnchant(theItem))` in the outer if statement prevent this block of code being run if the monster isn't stopped early in its path?